engine/Controller/SoundEngine.h

```diff
@@ -6,111 +6,37 @@
-*	@version 1
-*	@author Dylan Burgess
-*	@date 07/06/2023
-*/
-
 class SoundEngine
 {
 public:
-	/**
-	*	@brief default constructor
-	*/
 	SoundEngine();
 
-	/**
-	*	@brief gets the sound engine singleton
-	*	@return SoundEngine
-	*/
 	static SoundEngine& Get();
-	/**
-	*	@brief adds sounds to the sound map
-	*   @param name - the key for the map
-	*   @param filePath - location of the sound
-	*	@return void
-	*/
-	void addSound(std::string name, std::string filePath);
 
-	/**
-	*	@brief plays a sounds from the sound map
-	*   @param name - the sound to play
-	*   @param pos - position of the sound to play
-	*	@return void
-	*/
-	void playSound(std::string name, glm::vec3 pos);
+	void playSoundFromFile(std::string audioFilePath);
 
-	/**
-	*	@brief adds music to the sound engine
-	*	@param filePath - location of the music
-	*	@return void
-	*/
-	void addMusic(std::string filePath);
-	/**
-	*	@brief set the volume of the sound engine
-	*	@param volume the perctane of volume 0-1
-	*	@return void
-	*/
-	void setVolume(float volume);
+	void playSound(std::string audioName);
 
-	/**
-	*	@brief set the volume of the music engine
-	*	@param volume the perctane of volume 0-1
-	*	@return void
-	*/
-	void setMusicVolume(float volume);
+	void playLoop(std::string audioName);
 
-	/**
-	*	@brief set the position of the sound listener
-	*	@param pos - position of the listener
-	*	@return void
-	*/
-	void setListenerPos(glm::vec3 pos);
-	/**
-	*	@brief set the position of the music player
-	*	@param pos - position of the music player
-	*	@return void
-	*/
-	void setMusicPos(glm::vec3 pos);
+	void addSound(std::string audioName, std::string audioFilePath);
 
-	/**
-	*	@brief turns on or off the music
-	*	@param state - true for on / false for off
-	*	@return void
-	*/
-	void toggleMusic(bool state);
-	/**
-	*	@brief memory cleanup
-	*	@return void
-	*/
-	void drop();
+	void playSoundAtPosition(std::string audioName, glm::vec3 pos);
 
-	/**
-	*	@brief plays a looped sound
-	*	@param name - the sound to play
-	*	@param pos - position of the sound to play
-	* 	@param volume the perctane of volume 0-1
-	*	@return void
-	*/
-	void playSoundLoop(std::string name, glm::vec3 pos, float vol);
+	irrklang::ISound* playDynamicSound(std::string audioName, glm::vec3 pos);
 
+	void playLoopAtPosition(std::string audioName, glm::vec3 pos);
 
-	/**
-	*	@brief stops a looped sound
-	*	@param name - the sound to play
-	*	@return void
-	*/
-	void stopSoundLoop(std::string name);
+	void setUserPosition(glm::vec3 pos);
 
-private:
+	void setAudioPosition(irrklang::ISound* audio, glm::vec3 pos);
 
-	irrklang::ISoundEngine* _Engine;
-	std::map<std::string, std::string> _Audio;
-	irrklang::ISound* _Music;
+	std::vector<std::string> getAudioNames();
+private:
 
-	std::map<std::string, irrklang::ISound*> loopedAudio;
+	irrklang::ISoundEngine* engine;
+	std::vector<std::string> audioNames;
+	std::map<std::string, std::string> audio;
+	std::map<std::string, irrklang::ISound*> dynamicAudio;
```

Comment:
can we replace audio with a similar map? reading in audio from file everytime we want to play a sound seems pretty inefficient.

Reply:
It definitely does, but from what I've seen from irrklangs own tutorials and learn opengls audio tutorial you'll need to tell irrklang which file you want to play each time. An ISound is just a pointer to the function that allows you to reference an audio that is already playing so that you can manipulate the audio such as pausing, stopping, or changing position. But once an audio has stopped playing, an ISound does nothing and you'll have to read the file to play the same sound again.

Reply:
No problem, this should be fine then 👍